### ISSUE TITLE
Cloud Native High Availability 

### ISSUE THREAD

**Issue by hgoscenski-imanage** [jfrog/charts#1416]
**Is this a request for help?**: No

---

**Is this a BUG REPORT or FEATURE REQUEST?** (choose one): Bug Report

According a phone call with jfrog support (speaking to John Chiao and Jared Ledgister) the new Cloud Native HA should be supported in the latest version of the Helm Chart. 

We attempted to deploy it, by passing in a custom system.yaml in order to set the `taskAffinity` setting to `any` as specified in the documentation. When we deployed, we used just nodes/members with no primary, but received logs that indicated that the `taskAffinity` setting was being ignored. 

Looking at these 2 lines below it seems that the env variable to determine whether a node is a primary/member node is hard coded in the statefulsets and not configurable via the values file. Based on this, it does not seem possible to use the new cloud native ha with this current helm chart without stripping out the env variable prior to applying to the cluster.

https://github.com/jfrog/charts/blob/master/stable/artifactory-ha/templates/artifactory-primary-statefulset.yaml#L367
https://github.com/jfrog/charts/blob/master/stable/artifactory-ha/templates/artifactory-node-statefulset.yaml#L424

**Version of Helm and Kubernetes**:
```bash
[13:28:48] ~ λ kc version
hClient Version: version.Info{Major:"1", Minor:"21", GitVersion:"v1.21.0", GitCommit:"cb303e613a121a29364f75cc67d3d580833a7479", GitTreeState:"clean", BuildDate:"2021-04-08T21:16:14Z", GoVersion:"go1.16.3", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"19", GitVersion:"v1.19.7", GitCommit:"8ab00ff68a1763b5a506a7073cb8e67b12dfbcd7", GitTreeState:"clean", BuildDate:"2021-03-10T23:40:01Z", GoVersion:"go1.15.5", Compiler:"gc", Platform:"linux/amd64"}

[13:28:51] ~ λ helm version
version.BuildInfo{Version:"v3.5.4", GitCommit:"1b5edb69df3d3a08df77c9902dc17af864ff05d1", GitTreeState:"dirty", GoVersion:"go1.16.3"}
```

**Which chart**:
```bash
[13:15:59] ~ λ helm ls --kube-context ravn-dev-uk -n jfrog
NAME          	NAMESPACE	REVISION	UPDATED                               	STATUS  	CHART               	APP VERSION
jfrog-platform	jfrog    	2       	2021-04-15 18:03:59.738366 +0100 +0100	deployed	jfrog-platform-0.4.1	7.17.5
```

**What happened**:
Attempting to create a Cloud Native HA cluster using a custom system yaml, however it was not able to do so.

**What you expected to happen**:
Cloud native HA to have clear documentation and an example values.yaml file to start with such that it is easy to implement as a customer.

**Comment by chukka** [comment]
@hgoscenski-imanage This is a known issue. Will be fixed along with 7.18.x release  (this week) chart release.

**Comment by hgoscenski-imanage** [comment]
@chukka should I leave this issue open, or is there another on tracking this?

**Comment by chukka** [comment]
Please keep it open , Once chart is released, will close it

**Comment by hgoscenski-imanage** [comment]
Awesome, thank you for the prompt response.

**Comment by chukka** [comment]
@hgoscenski-imanage we identified an issue , CNHA was not working and Fix will be available in the next chart release of 7.18.x
